test/test_dataset.py

```diff
@@ -34,5 +35,14 @@ def test_listdirfiles_iterable_dataset(self):
         for pathname in dataset:
             self.assertTrue(pathname in self.temp_files)
 
+    def test_loadfilesfromdisk_iterable_dataset(self):
+        temp_dir = self.temp_dir.name
+        dataset1 = ListDirFilesIterableDataset(temp_dir, '')
+        dataset2 = LoadFilesFromDiskIterableDataset(dataset1)
+
+        for rec in dataset2:
```

Comment:
Are you sure that traverse order is guaranteed?

Reply:
Traverse order is guaranteed in LoadFilesFromDiskIterableDataset.
I think the question is whether the traverse order is guaranteed in ListDirFilesIterableDataset.
ListDirFilesIterableDataset use os.walk, currently no enforced order, should we enforce some kind of order? sort by file path+name?

Reply:
No need to order it.